Source/WebCore/platform/graphics/FontCascade.cpp

```diff
@@ -224,12 +224,7 @@ std::unique_ptr<DisplayList::DisplayList> FontCascade::displayListForTextRun(Gra
     ASSERT(!context.paintingDisabled());
     unsigned destination = to.value_or(run.length());
 
-    // FIXME: Use the fast code path once it handles partial runs with kerning and ligatures. See http://webkit.org/b/100050
-    CodePath codePathToUse = codePath(run);
-    if (codePathToUse != CodePath::Complex && (enableKerning() || requiresShaping()) && (from || destination != run.length()))
-        codePathToUse = CodePath::Complex;
```

Comment:
The duplicated conditional inside of `codePath` was harder to parse than it really should be so it would be nice if we could rewrite it at some point

Reply:
This changes non-auto behavior. I don't think that's expected.

Reply:
Yeah I agree I don't think that's the intention of the change and should be addressed in some way